src/System.Linq/tests/ConcatTests.cs

```diff
@@ -525,5 +525,66 @@ public static IEnumerable<object[]> GetToArrayDataSources()
                 }
             };
         }
+
+        [Theory]
+        [MemberData(nameof(GetConcatOptimizationDataSources))]
+        public void CollectionConcatenationOfEmptyOptimized(ConcatOptimizationCase[] testCases)
+        {
+            foreach (var a in testCases)
+            {
+                foreach (var b in testCases)
+                {
+                    var concatResult = a.Value.Concat(b.Value);
```

Comment:
Style: `var` is only allowed for `new`, `as`-cast, and hard-cast assignment statements.